### DISCUSSION TITLE
FTDI Pin Layout

### DISCUSSION THREAD

**Discussion by PrutzerNL** [SmartEVSE/SmartEVSE-2#38] [Q&A]
Hi All,

I have to update my Smart EVSE 2 to the latest firmware so I can use the Rfid reader.

Is this the right FTDI pinlayout?

<html>
<body>

Pin | Color | Function
-- | -- | --
1 | Black | GND - Ground
2 | Brown | CTS - Clear to Send
3 | Red | VCC- Positive Voltage Output (+5V)
4 | Orange | TXD- Transmit Asynchronous Data output
5 | Yellow | RXD- Receive Asynchronous Data input
6 | Green | RTS- Request to Send

</body>
</html>

I will set the USB to TTL adapter to 3V3 so the datalines are 3.3V. The 5V will stay 5V. Is that alright?
Power to the SmartEVSE on, Car disconnected.

**Comment by mstegen** [comment]
Yes the pinout is correct. On the SmartEVSE only the GND, TXD and RXD are used.
Please be careful for static discharge when connecting to these pins. It's best to first connect the GND, then the TXD and RXD. and then connect the USB plug to the PC.

**Comment by PrutzerNL** [comment]
Vandaag heb ik een poging gewaagt om de firmware te updaten.
Ik heb de smart EVSE uit de EVbox gehaald en een pinheader gesoldeerd.
De computer sluit ik aan met een FT232RL. Pin 1 aarde, pin 4 TX, pin 5 RX. (TX en RX ook al een keer gewisseld)
(Als eerste aarde verbonden). De FT232RL staat ingesteld op 3V3.

De evse voorzie ik van stroom en die gaat netjes aan. USB kabel in de computer en de TTL adapter wordt netjes gevonden. Com 3. Baud rate op 115200 gezet in computer beheer. 8N1.
Vervolgens AN1310 gestart. Ook hier de baud rate op 115200 gezet en de opties flash eeprom memory en EEProm aangezet.
Als ik nu op de stop knop druk geeft hij aan: No bootloader found.

Ik heb ook een andere TTL adapter (CP2102) geprobeerd. zelfde resultaat.
De volgende opties werken ook niet:

Andere laptop
Baud rate terug schroeven naar een lagere snelheid
Applicatie als Admin uitvoeren

Beide TTL USB adapters werken als ik ze voor andere toepassingen gebruik.
Wat doe ik fout?

**Comment by PrutzerNL** [comment]
Okee: Oplossing: Pinnen gaan van rechts naar links...

Het is dus  5 4 .... 1

De nieuwe firmware staat er op :)

**Reply by mstegen** [reply]
Mooi dat het gelukt is !
